rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,189 @@
+KEY_INTENT_TEXT = "text"
+KEY_SYNONYM = "synonym"
+KEY_SYNONYM_EXAMPLES = "examples"
+KEY_REGEX = "regex"
+KEY_REGEX_EXAMPLES = "examples"
+KEY_LOOKUP = "lookup"
+KEY_LOOKUP_EXAMPLES = "examples"
+
+DOCS_LINK = "https://rasa.com/docs/rasa/"
+
+
+class RasaYAMLReader(TrainingDataReader):
+    def __init__(self) -> None:
+        self.training_examples = []
+        self.entity_synonyms = {}
+        self.regex_features = []
+        self.lookup_tables = []
+
+    def reads(self, s: Text, **kwargs: Any) -> "TrainingData":
+        from rasa.nlu.training_data import TrainingData
+
+        self.__init__()
+
+        yaml_content = rasa.utils.io.read_yaml(s)
+
+        for key, value in yaml_content.items():
+            if key == KEY_NLU:
+                self._parse_nlu(value)
+            else:
+                logger.warning(
+                    f"Unexpected key '{key}' found in {self.filename}"
+                    f"Acceptable keys are: {KEY_NLU}"
+                )
+
+        return TrainingData(
+            self.training_examples,
+            self.entity_synonyms,
+            self.regex_features,
+            self.lookup_tables,
+        )
+
+    def _parse_nlu(self, nlu_data: List[Dict[Text, Any]]) -> None:
+
+        for nlu_item in nlu_data:
+            if not isinstance(nlu_item, dict):
+                logger.warning(
+                    f"Unexpected block found in '{self.filename}': \n"
+                    f">> {nlu_item}\n"
+                    f"Items under the `nlu` key must be YAML dictionaries."
+                    f"This block will be skipped."
+                    f"Please check the docs at {DOCS_LINK}."
+                )
+                continue
+
+            if KEY_INTENT in nlu_item.keys():
+                self._parse_intent(nlu_item)
+            elif KEY_SYNONYM in nlu_item.keys():
+                self._parse_synonym(nlu_item)
+            elif KEY_REGEX in nlu_item.keys():
+                self._parse_regex(nlu_item)
+            elif KEY_LOOKUP in nlu_item.keys():
+                self._parse_lookup(nlu_item)
+
+    def _parse_intent(self, data: Dict[Text, Any]) -> None:
+        from rasa.nlu.training_data import Message
+
+        intent = data.get(KEY_INTENT, "")
+        examples = data.get(KEY_INTENT_EXAMPLES, "")
+
+        for example, entities in self._parse_training_examples(examples):
+
+            SynonymsParser.add_synonyms_from_entities(
+                example, entities, self.entity_synonyms
+            )
+
+            plain_text = EntitiesParser.replace_entities(example)
+
+            message = Message.build(plain_text, intent)
+            message.set("entities", entities)
+            self.training_examples.append(message)
+
+    def _parse_training_examples(self, examples: Text) -> List[Tuple[Text, List[Dict]]]:
+
+        if isinstance(examples, list):
+            iterable = [
+                example.get(KEY_INTENT_TEXT, "") for example in examples if example
+            ]
```

Comment:
I was thinking that if the user specifies a list of strings, like:
```yml
examples:
  - hello
  - hi there
  - heyy
```
Could we accept that as well? Or maybe we want to be stricter and support less variants. In that case, the code here errors out because we try to call `.get()` on a string.

Reply:
Looks like I've written the code here exactly for the "list of strings" use case! I'll clarify if we want to have both.

Reply:
see https://github.com/RasaHQ/rasa/pull/5989#issuecomment-651910420